composer/loggers/logger_hparams.py

```diff
@@ -215,3 +216,27 @@ class TQDMLoggerBackendHparams(BaseLoggerBackendHparams):
     def initialize_object(self, config: Optional[Dict[str, Any]] = None) -> TQDMLoggerBackend:
         from composer.loggers.tqdm_logger import TQDMLoggerBackend
         return TQDMLoggerBackend(config=config)
+
+
+@dataclass
+class MosaicMLLoggerBackendHparams(BaseLoggerBackendHparams):
+    """:class:`~composer.loggers.mosaicml_logger.MosaicMLLoggerBackend`
+    hyperparameters.
+
+    See :class:`~composer.loggers.mosaicml_logger.MosaicMLLoggerBackend`
+    for documentation.
+    """
+    job_id: int = hp.required("The id of the job to write logs for.")
+    sweep_id: int = hp.required("The id of the sweep to write logs for.")
+    creds_file: Optional[int] = hp.optional(
+        "A file containing the MosaicML api_key. If not provided "
+        "will default to the environment variable MOSAIC_API_KEY",
+        default=None)
+    flush_every_n_batches: int = hp.optional("Flush the log data buffer every n batches.", default=100)
```

Comment:
This is outside the scope of the PR, but we should probably have a global flush interval hparam, so there is a lot of flushing happening in a bunch of places.

Reply:
Yup - we can consolidate in a separate PR